src/graphql/graphql.ts

```diff
@@ -962,16 +977,28 @@ export function buildIncrementalQueryV1(type: gql.GraphQLObjectType): Query {
     } else if (isV1ModelType(field.type)) {
       // this is foreign key to a top-level model.
       // add nested fragment to select id of referenced model
-      let nestedName = `${field.name}Id`;
-      // check for collision between nested name and scalars
-      if (_.has(type.getFields(), nestedName)) {
-        nestedName = `${field.name}Fk`;
+      if (avoidCollisions) {
+        let nestedName = `${field.name}Id`;
+        // check for collision between nested name and scalars
+        if (_.has(type.getFields(), nestedName)) {
+          nestedName = `${field.name}Fk`;
+        }
+        fieldsObj[field.name] = {
+          [`${nestedName}: id`]: true,
+        };
+      } else {
+        fieldsObj[field.name] = {
+          [ID_FLD]: true,
+        };
       }
-      fieldsObj[field.name] = {
-        [`${nestedName}: id`]: true,
-      };
     }
   }
+
+  // Add refreshedAt
+  fieldsObj.metadata = {
+    refreshedAt: true,
```

Comment:
what does it mean `refreshedAt: true`?

Reply:
I think the `true` is arbitrary here. The goal is to include the query selection set in `fieldsObj`.

Kudos to @ted-faros for making this easy to extend

Reply:
+1 to what @ypc-faros said about it being arbitrary.  We just trying to build a list and then [transform with json-to-graph-query](https://github.com/vkolgi/json-to-graphql-query#simple-query).